### DISCUSSION TITLE
How can I invalidate sessions when a user's password changes?

### DISCUSSION THREAD

**Discussion by jerrac** [keycloak/keycloak#33210] [Q&A]
How can I use the admin UI to tell Keycloak that if a user's credentials change, the user should have their browser sessions terminated? (I've been searching and I have not found a way yet, which makes me wonder if it is even possible...)

I have found the REST api docs for sessions, but I'd think this kind of functionality would be part of Keycloak itself, at least for users stored in Keycloak's db. So I thought I'd ask before I go trying to use the API for this.

**Comment by codespearhead** [comment]
Are you using the latest version of Keycloak (25.0.6)?

Because this looks like a duplicate of https://github.com/keycloak/keycloak/issues/10232 , which was resolved by https://github.com/keycloak/keycloak/pull/21897 a year ago.

**Reply by jerrac** [reply]
25.0.2 is what I'm on right now. 

That issue and the pull request are about setting an optional check box the user can check when they change their password, that will invalidate their sessions, right? 

I'm looking for something that forces the sessions to be invalidated with no user intervention. Specifically because our current password change workflow occurs outside of Keycloak. So the password update is pushed in via the api. (Which is why I was looking at the api as an option, I just would rather use something built into Keycloak if possible.)

**Reply by codespearhead** [reply]
Try the `POST /admin/realms/{realm}/users/{id}/logout` endpoint from the [Keycloak Admin REST API](https://www.keycloak.org/docs-api/22.0.1/rest-api/index.html):

> POST /admin/realms/{realm}/users/{id}/logout
Remove all user sessions associated with the user. Also send notification to all clients that have an admin URL to invalidate the sessions for the particular user.
